### ISSUE TITLE
accessibilityRole="link" is not focusable

### ISSUE THREAD

**Issue by calebmer** [necolas/react-native-web#1266]
**The problem**
An element with `accessibilityRole="link"`, like `<TouchableOpacity>`, is not focusable.

```jsx
<TouchableOpacity accessibilityRole="link" onPress={() => console.log("navigate")}>
  <Text>Focus Me!</Text>
</TouchableOpacity>
```

**How to reproduce**

Simplified test case: https://codesandbox.io/s/w2n7514m9l

Steps to reproduce:
1. Focus the first `<TouchableOpacity accessibilityRole="button">`
2. Press tab.

**Expected behavior**

The focused element should be the second `<TouchableOpacity accessibilityRole="link">` but instead focus is given to the third `<TouchableOpacity accessibilityRole="button">`.

**Environment**

- Browser: Chrome
- OS: MacOS

**Workaround**

For now, my workaround is to use:

```jsx
<TouchableOpacity
  accessibilityComponentType="a"
  accessible={true}
  onPress={() => console.log("navigate")}
>
  <Text>Focus Me!</Text>
</TouchableOpacity>
```

**Fix?**

Here:

https://github.com/necolas/react-native-web/blob/eb7706260813f843a9d7d7ad90e2d612d11f867f/packages/react-native-web/src/modules/createDOMProps/index.js#L123-L125

It’s assumed that `<a>` is focusable by default. Based on basic testing that might not be true? Not sure how this is specced.

**Comment by paularmstrong** [comment]
It looks like the tabindex is getting ignored because the buttons surrounding the link have `tabindex="0"` ([source](https://github.com/necolas/react-native-web/blob/eb7706260813f843a9d7d7ad90e2d612d11f867f/packages/react-native-web/src/modules/createDOMProps/index.js#L144)) but links do not receive a `tabIndex` ([source](https://github.com/necolas/react-native-web/blob/eb7706260813f843a9d7d7ad90e2d612d11f867f/packages/react-native-web/src/modules/createDOMProps/index.js#L139)). 

According to [this resource](https://webaim.org/techniques/keyboard/tabindex), if you set a tabindex on some elements, even if `0`, they will be ordered first, before links that do not have `tabindex` explicitly set.

**Comment by calebmer** [comment]
One thing to note, if I add an `href` to the element it’s focusable. e.g.

```jsx
<TouchableOpacity
  accessibilityRole="link"
  href="#"
  onPress={() => console.log("navigate")}
>
  <Text>Focus Me!</Text>
</TouchableOpacity>
```

This leads me to believe that `<a href>` is focusable but `<a>` is not.

**Comment by paularmstrong** [comment]
> A value of 0 indicates that the element should be placed in the default navigation order. This allows elements that are not natively focusable (such as <div>, <span>, <p>, and <a> with no href) to receive keyboard focus.

This means that with no `href`, it's intended that it is not focusable.

**Comment by giuseppeg** [comment]
> with no href, it's intended that it is not focusable.

ditto

**Comment by calebmer** [comment]
So one should not use `accessibilityRole="link"` if you want a focusable element? Maybe this should be documented?

**Comment by necolas** [comment]
If you want the link to be focusable (and accessible) on web add an `href`

**Comment by rveruna** [comment]
Yes, I just 'faked' the href, as the linking works with or without it. and no matter what you put in it
